Docs: Fundamentals of Block Development - File structure of a block (closed)

juanmaguitar opened the pull request:
Add documentation for file structure of a block
This PR closes the issue https://github.com/WordPress/gutenberg/issues/55437

juanmaguitar committed:
Add file structure of a block documentation

juanmaguitar committed:
Add file structure of a block documentation

carolinan commented on docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md:
```suggestion
The recommended way to register a block is via a plugin. With the `create-block` tool you can quickly scaffold the structure of the files required to create a plugin that registers a block. 
```

juanmaguitar committed:
Update docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md

Thanks @carolinan

Co-authored-by: Carolina Nymark <myazalea@hotmail.com>

juanmaguitar committed:
Update docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md

Thanks @carolinan

Co-authored-by: Carolina Nymark <myazalea@hotmail.com>

carolinan commented on docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md:
I found this difficult to read. Does this describe block.json or the list of files below the text? I think I struggled because of the plural form.

carolinan commented on docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md:
Will it be clear enough that this is a scss / sass file, or do we need to add something more about the file format and the processing?

fabiankaegy commented on docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md:
This is incorrect. At its simplest it is a registerBlockType function call with some parameters passed in JS

Block.json is just an abstraction that makes this easier and allows us to more easily share the configuration between JS and PHP.

fabiankaegy commented on docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md:
Again this file is not required at all :). 

It just makes things easier.

fabiankaegy commented on docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md:
In the viewScript property

youknowriad committed:
Extract a PostPanelRow component from the different sidebar panels (#56238)

juanmaguitar committed:
Update page as per feedback received

juanmaguitar committed:
Update page as per feedback received

juanmaguitar commented on docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md:
I have updated this part taking into account the  feedback of both you 🙏

juanmaguitar commented on docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md:
I have added an info box providing some more context about these `.scss` files 👍

juanmaguitar commented on docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md:
Thanks a lot for this feedback @fabiankaegy 🙌
I have adjusted the sentences in this page to take this into account.

juanmaguitar committed:
block.json as common way to register a block

juanmaguitar committed:
block.json as common way to register a block

juanmaguitar commented:
Thanks a lot for your feedback @carolinan and @fabiankaegy 🙏
I have updated the PR addressing your comments.

juanmaguitar committed:
Fix code formatting

juanmaguitar committed:
Fix code formatting

juanmaguitar committed:
Remove unnecessary explanation at this stage

juanmaguitar committed:
Remove unnecessary explanation at this stage

juanmaguitar committed:
Added clarifications and examples to "Get started with wp-scripts" (#56298)

juanmaguitar committed:
Link preview image to live example using WordPress Playground (#56292)

* Embed preview of the example using iframe and WordPress Playground

Embed preview of the example using iframe and WordPress Playground

* Add link to live preview

* Update docs/how-to-guides/data-basics/README.md

Co-authored-by: Alex Stine <alex.stine@yourtechadvisors.com>

* proper alt and titles attributes added

* proper alt and titles attributes added

* Update docs/how-to-guides/data-basics/README.md

Co-authored-by: Alex Stine <alex.stine@yourtechadvisors.com>

---------

Co-authored-by: Alex Stine <alex.stine@yourtechadvisors.com>

fabiankaegy commented on docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md:
```suggestion
The most common way to register a custom block is via a JSON file that contains all of the metadata of the block.
```

fabiankaegy commented on docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md:
```suggestion
It is recommended to register blocks within plugins to ensure they stay available when a theme gets switched. With the `create-block` tool you can quickly scaffold the structure of the files required to create a plugin that registers a block. 
```

fabiankaegy commented on docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md:
```suggestion
This file contains the metadata of the block, and it's used to simplify the definition and registration of the block both in the browser and on the server. 
```

fabiankaegy commented on docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md:
```suggestion
The `index.js` file (or any other file defined in the `editorScript` property of `block.json`) is the entry point file for javascript that should only get loaded in the editor. It is responsible for calling the `registerBlockType` function to register the block on the client. In a standard structure it imports the `edit.js` and `save.js` files to get functions required in block registration.
```

fabiankaegy commented on docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md:
```suggestion
The `edit.js` commonly gets used to contain the React component that gets used in the editor for our block. It usually exports a single component that then gets passed to the `edit` property of the `registerBlockType` function in the `index.js` file.
```

fabiankaegy commented on docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md:
```suggestion
The `save.js` is similar to the `edit.js` file in that it exports a single React component. This component generates the static HTML markup that gets saved to the Database. 
```

fabiankaegy commented on docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md:
I think it is a little missleading to only talk about SCSS here. WP scripts also supports regular straight up CSS and postCSS here.

fabiankaegy reviewed (commented):
I added some more inline suggestions :)

jasmussen committed:
Try: Change "Detach pattern" to "Detach" (#56323)

* Update: Change "Detach patterns" to singular.

* Try: "Detach" instead of "Detach pattern".

juanmaguitar committed:
Update docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md

Co-authored-by: Fabian Kägy <mail@fabian-kaegy.de>

juanmaguitar committed:
Update docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md

Co-authored-by: Fabian Kägy <mail@fabian-kaegy.de>

juanmaguitar committed:
Update docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md

Co-authored-by: Fabian Kägy <mail@fabian-kaegy.de>

juanmaguitar committed:
Update docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md

Co-authored-by: Fabian Kägy <mail@fabian-kaegy.de>

juanmaguitar committed:
Update docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md

Co-authored-by: Fabian Kägy <mail@fabian-kaegy.de>

juanmaguitar committed:
Update docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md

Co-authored-by: Fabian Kägy <mail@fabian-kaegy.de>

juanmaguitar committed:
Update docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md

Co-authored-by: Fabian Kägy <mail@fabian-kaegy.de>

juanmaguitar committed:
Update docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md

Co-authored-by: Fabian Kägy <mail@fabian-kaegy.de>

juanmaguitar committed:
Update docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md

Co-authored-by: Fabian Kägy <mail@fabian-kaegy.de>

juanmaguitar committed:
Update docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md

Co-authored-by: Fabian Kägy <mail@fabian-kaegy.de>

juanmaguitar committed:
Update docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md

Co-authored-by: Fabian Kägy <mail@fabian-kaegy.de>

juanmaguitar committed:
Update docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md

Co-authored-by: Fabian Kägy <mail@fabian-kaegy.de>

fabiankaegy commented:
@juanmaguitar Thanks for your patience here by the way :) I feel like I'm being really nitpicky today... Loving all these documentation updates you are working on currently!

juanmaguitar committed:
Included several format of CSS

juanmaguitar committed:
Included several format of CSS

juanmaguitar commented on docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md:
Thanks for this feedback @fabiankaegy  👌
I have expanded the explanation to include more CSS formats accepted by `wp-scripts`. 
Let me know what you think

juanmaguitar commented:
> @juanmaguitar Thanks for your patience here by the way :) I feel like I'm being really nitpicky today... Loving all these documentation updates you are working on currently!

@fabiankaegy No worries at all! I love your comments and feedback. Please, keep them coming 🙂
They're really helping shape these docs and make them much more accurate.

fabiankaegy commented on docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md:
```suggestion
This file contains the metadata of the block, and it's used to simplify the definition and registration of the block both in the client and on the server. 
```

I made this mistake here. I much prefer how you used `client` later on in the document :)

juanmaguitar committed:
Update docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md

Co-authored-by: Fabian Kägy <mail@fabian-kaegy.de>

youknowriad committed:
Post Schedule Panel: Remove text overflow ellipsis (#56319)

t-hamano committed:
Duotone: Backport from Core to fix filters in classic themes (#54778)

Co-authored-by: Alex Lende <alex+github.com@lende.xyz>

juanmaguitar committed:
Docs - Block Editor Handbook - Add end user documentation about Block Editor as a resource on the Landing Page of the Block Editor Handbook (#49854)

* add end user documentation as a resource

* Update docs/README.md

Co-authored-by: Nick Diego <nick.diego@automattic.com>

* Moved End User Documentation to the end of the list

---------

Co-authored-by: Nick Diego <nick.diego@automattic.com>

juanmaguitar committed:
Update webpack options for wp-scripts in README.md (#56314)

* Update webpack options in README.md

* Add support for custom output directory in build
process

ramonjd committed:
Revisions: add new selectors to fetch entity revisions (#54046)

* Fetching entity revisions using loaded name argument e.g., post:id:revisions

Revisions loaded into state tree under revisions[parent]

Invalidate revisions after saved entity

Revisions url params are contained in entity  config

Testing with global styles - modifying CPT to show in rest

Implementing thin API layer for Revisions over the getEntityRecords implementation

Duplicated the logic in the selectors for revisions functions

Added tests

* Revert edit context query param when fetching post types
Reverting timeout in performance tests.

* Tidying up reducer, allowing revisions for posts and pages only

* Adding types to the base entity records for revisions and global styles revisions.

* Removed unnecessary REST API change for global styles: it was to pull in the CPT in the /types response, which we don't need for core-data since it's a root entity.

* Remove duplicated queried-data reducer functions for revisions, and use them in the main reducer

Create dedicated resolvers for getRevision/s, thereby removing the split name parsing required (name:id:revisions)
Create dedicated action receivers for revisions
Rename from getEntityRevisions to getRevisions

Fully remove parseEntityName

Updated selector tests

Use queriedDataReducer shape for revisions. The queriedDataReducer doesn't need to know about the revisions action type.

Update doc comments

Updated CHANGELOG.md

* This commit:
- removes `hasRevisions` because it has no noticeable benefit, at least as far as I can see. I'm probably wrong.
- removes the default revisions query context for the API call (the default is 'view' anyway)
- fixes the revisions reducer
- Add _fields logic to resolvers and selectors

Adds basic reducer tests

* updated variable names: parentId > recordKey, key > revisionKey
removing lock-related code
deleting `recordKey` from action object before sending to queriedDataReducer
updating docs

* Removing revisions when items are removed.

Commit:
Migrating `BlockPatternSetup` to use updated `Composite` implementation (#55425)

 - Removes `__unstableComposite` imports from `@wordpress/components`
 - Adds private `Composite*` exports from `@wordpress/components`
 - Refactors `BlockPatternSetup` and subcomponents to use updated `Composite` components
 - Additionally adds image semantics, and ensures buttons are available when disabled

andrewserong committed:
useInnerBlocksProps: stabilise dropZoneElement prop (#56313)

ellatrix committed:
Rich text: pad multiple spaces through en/em replacement (#56341)

ramonjd committed:
Site editor: redirect to main page menu if page record not found (#56177)

* Testing redirecting to /page when a record has resolved but has not been found, probably deleted.

* Removed unused var `isCanvasModeView`

Commit:
Update README.md to fix the typo (#56322)

Before: onKeydown
After: onKeyDown

t-hamano committed:
Docs: Fix incorrect build script description in script package (#56332)

arthur791004 committed:
Block Theme Preview: Display the theme name on the activate button (#55752)

* Block Theme Preview: Display the theme name on the activate button

* Fix code styles

draganescu committed:
Add top toolbar to distraction free mode (#56295)

Mamaduka committed:
Migrate 'sidebar permalink' e2e tests to Playwright (#56253)

* Migrate 'sidebar permalink' e2e tests to Playwright

Co-authored-by: Pavan  Patil <44057535+pavanpatil1@users.noreply.github.com>

* Remove old test file

* Open document settings sidebar

---------

Co-authored-by: Pavan  Patil <44057535+pavanpatil1@users.noreply.github.com>

carolinan commented on docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md:
Isn't block.json required to be able to style the block with theme.json? If only I could recall exactly where I got that information from. I assume it is related to API versions...

fabiankaegy commented on docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md:
@carolinan It may be the fact that a block needs to be registered on the server for global styles for the block to get printed correctly. 


But you can register a block on the server without using the `block.json` file. 

The file just makes it easier :)

ntsekouras committed:
Dataviews: Ensure items and fields are using a unique id (#56366)

youknowriad committed:
Site Editor: Add the Post Author component to the Page sidebar (#56368)

fabiankaegy committed:
Fix version in useSetting deprecation notice (#56377)

ramonjd committed:
Global styles revisions: migrate API call to getRevisions()  (#56349)

* Migrating global styles revisions to new core data API and deprecating *ThemeGlobalStyleRevision selector and action.

* Limiting the results to 100

glendaviesnz committed:
Spacing theme.json - check that the first item in the units array is defined before trying to assign its value (#56306)

glendaviesnz committed:
Patterns: Fix bug with templates in replace template screen showing as synced patterns (#56407)

andrewserong committed:
useMovingAnimation: Clear translate3d rule when animation is finished (#56410)

* useMovingAnimation: Clear translate3d rule when animation is finished

* Add inline comment for clarity

tellthemachines committed:
Reduce specificity of default Cover text color styles. (#56411)

Mamaduka committed:
Pattern placeholder: Remove duplicate 'useDispatch' hook (#56397)

fabiankaegy committed:
Fix overly complex code example in ComboboxControl readme (#56365)

Mamaduka committed:
Block Editor: Undeprecate the '__experimentalImageSizeControl' component (#56414)

fabiankaegy committed:
Simplify code example in ToggleControl component readme (#56389)

jameskoster committed:
Fix data view menu item radius (#56395)

* Fix data view menu item radius

* Update packages/edit-site/src/components/sidebar-dataviews/style.scss

* fix wrong GH suggestion

---------

Co-authored-by: Nik Tsekouras <ntsekouras@outlook.com>

gutenbergplugin committed:
Bump plugin version to 17.1.0

Commit:
Migrating `InserterListbox` to use updated Composite implementation (#56246)

 - Removes `__unstableComposite` imports from `@wordpress/components`
 - Adds private `Composite*` exports from `@wordpress/components`
 - Refactors `InserterListbox`, `InserterListboxRow` and `InserterListboxItem` to use updated `Composite` components
 - Additionally updates `BlockTypesList` to ensure listbox items are appropriately grouped

gutenbergplugin committed:
Update Changelog for 17.1.0

oandregal committed:
DataViews: update filter component (#56110)

bph committed:
Update Readme.txt tested up to 6.4 (#56427)

* fixes wrong link in #56084

* update 'tested up to' in readme.txt of GB plugin

draganescu committed:
Add align support to the image block - alternative (#55954)

* only remove the alignment control

* use an effect to reset width, height, aspect ratio and cover when align is wide or full

* try a filter on block attributes when align control updates attributes

* new fixtures for the default align attribute of core/image

* updated snapshots

* modify the filter to always return the same shape of object, update naming

* remove filter, add undo skip

* update the react native component

* Remove align from custom image attributes

This allows the block supports to add the attribute instead keeping things cleaner.

* Fix tests

* Fix another fixture

---------

Co-authored-by: Alex Lende <alex+github.com@lende.xyz>

fabiankaegy committed:
fix remove removal version from deprecated prop (#56336)

t-hamano committed:
NavigableContainers: Fix doc typo in onKeyDown prop (#56352)

* NavigableContainers: Fix doc typo in onKeyDown prop

* Update changelog

* Remove changelog

MaggieCabrera committed:
Improve tooltip for parent blocks on the block toolbar (#56146)

* improve tolltip for parent blocks on the block toolbar

* fixed tests

* missing test

* another missing test

* fixed another test

zaguiini committed:
PostCSS style transformation: fail gracefully instead of throwing an error (#56093)

* PostCSS style transformation: fail gracefully instead of throwing error

* Display better warning messages

youknowriad committed:
Documentation: Add the attributes definition page to the create block tutorial of the platform docs (#56429)

Co-authored-by: Héctor <27339341+priethor@users.noreply.github.com>

t-hamano committed:
Theme JSON schema: add heading/button key to color definition (#55674)

* Theme JSON schema: add heading/button key to color definition

* Update docs

* Add "in a block"

Mamaduka committed:
Playwright Utils: Fix the method of getting post ID in 'publishPost' (#56421)

* Playwright Utils: Fix the method of getting post ID in 'publishPost'
* Update sidebar permalink e2e test

oandregal committed:
Documentation: add new section to release docs about troubleshooting the release (#56436)

Commit:
Migrating block editor `BlockPatternsList` component (#56210)

 - Removes `__unstableComposite` imports from `@wordpress/components`
 - Adds private `Composite*` exports from `@wordpress/components`
 - Refactors `BlockPatternsList` and `BlockPattern` to use updated `Composite` components
 - Additionally renames list component to be consistent with codebase

ciampo committed:
Slot: add styles prop to bubblesVirtually version (#56428)

* Slot: add `styles` prop to `bubblesVirtually` version

* CHANHGELOG

* Update CHANGELOG

* Enhance clarity of JSDoc comments

Commit:
Add focus rings to focusable disabled buttons (#56383)

Fixes the CSS to only remove borders from disabled secondary/tertiary buttons when they don't have focus

zaguiini committed:
Additional CSS: fix on change validation (#56434)

Mamaduka committed:
Block Settings: Only display parent block selector on small screens (#56431)

jameskoster committed:
Add 'View site' action to 'Site updated' snackbar (#52693)

* Add 'View site' action to 'Site updated' snackbar

* Both Snackbars...

gutenbergplugin committed:
Update Changelog for 17.0.3

andrewserong committed:
Post Schedule Panel: Fix Sass deprecation warning for division (#56412)

t-hamano committed:
Post Template: fix incorrect offset query (#56440)

* Post Template: fix incorrect offset query

* Remove unnecessary code

tellthemachines committed:
Fix fatal error when calling undefined block library function. (#56459)

gutenbergplugin committed:
Bump plugin version to 17.1.1

gutenbergplugin committed:
Update Changelog for 17.1.1

andrewserong committed:
Cover block: Pass dropZoneElement reference to fix dragging within cover block area (#56312)

* Cover block: Pass dropZoneElement reference to fix dragging and dropping within cover block area

* Move to using a simple ref

youknowriad committed:
Move the DisableNonContentBlocks component to the editor package (#56423)

Mamaduka committed:
Block Editor: Optimize 'Connections' inspector controls (#56443)

fluiddot committed:
[RNMobile] Fix ungroup functionality in `WarningMaxDepthExceeded` component (#56445)

ntsekouras committed:
Dataviews: Remove link from author (#56467)

Mamaduka committed:
useEntityRecord: Improve unit tests (#56415)

* useEntityRecord: Improve unit tests

Co-authored-by: Kai Hao <kevin830726@gmail.com>

* Add comment

---------

Co-authored-by: Kai Hao <kevin830726@gmail.com>

ntsekouras committed:
DataViews: Make disabled pagination buttons focusable (#56422)

ciampo committed:
DropdownMenu V2: add support for rendering in legacy popover slot (#56342)

* DropdownMenuV2: support rendering in slotfill

* Add temporary stories

* CHANGELOG

jorgefilipecosta committed:
DataViews: Code Quality remove some unused props from action. (#56477)

ntsekouras committed:
Dataviews: Add preview and grid view in templates list (#56382)

* Dataviews: Add preview and grid view in templates list

* Update preview design

* Preview heights in grid view

* fix linting issue

* address feedback

* minor tweak

---------

Co-authored-by: James Koster <james@jameskoster.co.uk>

afercia committed:
Tooltip: Don't render buttons tooltip when show button text labels is enabled (#55842)

* Don't render Preview button tooltip when show text labels is enabled.

* Don't render View button tooltip when show text labels is enabled.

* Fix comment block.

* Don't render View Post button tooltip when show text labels is enabled.

ciampo committed:
DropdownMenu V2 tweaks (#56041)

* Expose help text subcomponent (uses truncate under the hood)

* Remove DropdownMenuGroupLabel

* Submenu trigger always display chevron next to suffix

* Tweak gutter so that submenus overlap by 8px

* Implement auto-indentation via context

* Revert "Implement auto-indentation via context"

This reverts commit a1e2d3af0285f1f28d1677d41005bcdb25fd0659.

* Use CSS grid/subgrid to impement auto-indentation

* Fix slotfill example

* Tweak sizes, spacing, and colors. Tidy up styles.

* Add label subcomponent

* Update storybook examples

* Export help text and label subcomponents

* CHANGELOG

* README

* Fix unit test

* Allow disabled items to be keyboard focusable / accessible

* Display the "not allowed" cursor on disabled menu items

* Add scroll-margin to make sure that menu items are fully visible when scrolled into view

* Update unit tests

* Use inline styles instead of !important

* Apply design feedback

* CHANGELOG

* Update submenu shift

ciampo committed:
Revert "DropdownMenu V2: add support for rendering in legacy popover slot (#56342)" (#56484)

This reverts commit 0b6c4a003f90b18509ee6d3d120435444dc15e43.

jameskoster committed:
Simplify dataviews view button (#56485)

* Simplify dataviews view button

* remove prop that doesn't exist

---------

Co-authored-by: ntsekouras <ntsekouras@outlook.com>

SavPhill committed:
Update "All patterns" to "All" in block inserter (#55986)

* Update label

* Update utils.js with review change

youknowriad committed:
Site Editor: Fix template resolution for templates assigned as home page (#56418)

Co-authored-by: ntsekouras <ntsekouras@outlook.com>
Co-authored-by: Ramon <ramonjd@users.noreply.github.com>

youknowriad committed:
Performance: update base hash

brookewp committed:
Try Ariakit Select for new CustomSelectControl component (#55790)

* Create base component using ariakit

* Simplify story example

* Add options for children and value with stories

* Update types

* Add uncontrolled story and adapt value

* Remove duplicate import

* Add size prop

* Changes from pairing session with ciampo

* Rename and cleanup

* Update styles

* Add context

* Add README

* Update manifest

* Update types and move context provider

* Add back types for multi-selection

* Remove file for moved component

* Add translation for placeholder

* Add check and cleanup styles

* Add multiselect example

* Require type children for CustomSelectItem

* Add small size and update sizing logic

* Update stories with ciampo’s suggested changes

* Update styles to match legacy CustomSelectControl

* Cleanup based on PR feedback

* Children as optional

* Incorporate ciampo’s suggestions to refine styles

* Make value required for CustomSelectItem and fix naming in comment

* Update changelog

* Refine styles and add styling to active item

* Only render label if defined

* Cleanup story

* Add suggestions to WIP README and types

* Make label required

* Improve logic for default rendered value

* Update stories

Commit:
Small tweaks to CreatePatternModal (#56016)

gutenbergplugin committed:
Bump plugin version to 17.1.2

gutenbergplugin committed:
Update Changelog for 17.1.2

ramonjd committed:
Core data: harmonize getRevision selector and resolver function signatures (#56416)

* Remove default value of query {} on getRevision. It messes up selector memoization since the function signatures are different.

* Resolve individual revisions as per `getEntityRecords`. This ensures that any subsequent calls to `getRevision` for revisions that we've already fetched via `getRevisions()` are marked as resolved and therefore do not trigger the resolver (and therefore an API call).

* Copy update.
Now that https://github.com/WordPress/gutenberg/pull/55827 has landed we can start supporting individual global styles URLs as well.

* While I'm here... :)

tellthemachines committed:
Try updating base performance hash again.

ellatrix committed:
E2e tests: merge Puppeteer into single job, split Playwright further (#56363)

* E2e tests: merge Puppeteer into single job, split Playwright further

* Remove shard handling for Puppeteer

---------

Co-authored-by: George Mamadashvili <georgemamadashvili@gmail.com>

jameskoster committed:
Update data view menu item actions (#56398)

* Update data view menu item actions

* Small button size

* add specifc class to dropdown menu

* fix linting issue

---------

Co-authored-by: ntsekouras <ntsekouras@outlook.com>

ramonjd committed:
isResolving wasn't updated in https://github.com/WordPress/gutenberg/pull/56349 (#56491)

This change does that

Mamaduka committed:
Migrate 'custom taxonomies' e2e test to Playwright (#56486)

* Migrate 'custom taxonomies' e2e test to Playwright
* Remove old test file

---------

Co-authored-by: Alvi Tazwar <55917380+alvitazwar@users.noreply.github.com>

oandregal committed:
DataViews: scope names of V2 UI components (#56503)

youknowriad committed:
Documentation: Add thee block supports page to the create block tutorial of the framework docs. (#56483)

Co-authored-by: Andrew Serong <14988353+andrewserong@users.noreply.github.com>
Co-authored-by: Ramon <ramonjd@users.noreply.github.com>

oandregal committed:
DataViews: rename `InFilter` component to `FilterSummary` (#56506)

youknowriad committed:
Editor: Move the template focus modes to the editor store (#56472)

fluiddot committed:
Mobile Release v1.109.0 (#56475)

* Release script: Update react-native-editor version to 1.109.0

* Release script: Update CHANGELOG for version 1.109.0

* Release script: Update podfile

* Update `react-native-editor` changelog

* Update `react-native-editor` changelog

afercia committed:
Improve `Button` saving state accessibility. (#55547)

* Make saving buttons honor prefers-reduced-motion.

* Fix edit site save hub button color contrast.

* Add changelog entry.

* Add inline comment.

getdave committed:
Backmerge block renaming fixes/refactors from 6.4 branch into Gutenberg trunk (#56386)

* Apply refactor from 6.4 branch

* Apply fix to lock down rename

* Update support hook to check correct support value

* Rename hook to match feature

* Correct test name

* Assert against a Block that doesn’t support the feature

* Ensure presence of options menu before Rename presence assertion

* Avoid redundant creation of blocks and use Group directly

* Add additional test coverage for options menu on unselected blocks

* Update hook to check for correct support

Commit:
Migrating `StyleBook` to use updated `Composite` implementation (#55344)

- Removes __unstableComposite imports from @wordpress/components
- Adds private Composite* exports from @wordpress/components
- Refactors Examples/Example within StyleBook to use updated Composite components
- Additionally, adds semantics around Composite to make UX clearer for assistive technology users

ciampo committed:
Save post button: avoid extra re-renders when enablng/disabling tooltip (#56502)

* Save post button: avoid extra re-renders when enablng/disabling tooltip

* Remove unnecessary import

* Typos

* Remove leftover comment

* Apply feedback

brookewp committed:
`BoxControl`: Update story and refactor to Typescript (#56462)

* BoxControl: Update story and refactor to Typescript

* Cleanup and refine story based on feedback

* Add Uncontrolled and Controlled templates and Controlled example

* Remove default values and fix values control

NekoJonez committed:
Add translator comments for strings containing date formats (#56531)

ramonjd committed:
the editor store action `savePost` doesn't expect to be passed an event object. (#56529)

tellthemachines committed:
Remove compatibility layer for WP 6.2 (#56464)

* Remove compatibility layer for WP 6.2

* update min required version in doc

* Delete obsolete tests.

andrewserong committed:
List View: Expand state if a block is dragged to within a collapsed block in the editor canvas (#56493)

aaronrobertshaw committed:
Design Tools: Fix last ToolsPanelItem styling (#56536)

ramonjd committed:
Style engine: allow CSS var output for fontSize and fontFamily and update docs (#56528)

* - Loading style engine files that use utils classes after the utils classes have loaded
- Adding CSS var parsing capability to fontSize and fontFamily. The JS package does this, theme.json has these presets, why not the Style Engine? I think it wasn't a deliberate omission.
- Updating docs in relation to CSS var names

* Apply suggestions from code review

Spelling fix

Co-authored-by: Andrew Serong <14988353+andrewserong@users.noreply.github.com>

* Replacing spacing indents with tabs. Disclaimer: This commit does not necessarily reflect the views or positions of the author in relation to tabs vs spaces.

---------

Co-authored-by: Andrew Serong <14988353+andrewserong@users.noreply.github.com>

talldan committed:
Fix FormTokenField suggestions broken scrollbar when `__experimentalExpandOnFocus` is defined (#56426)

* Fix FormTokenField suggestions being unscrollable when `__experimentalExpandOnFocus` is defined

* Update focus types

* Add changelog entry

* Type the form token field onFocus event as a React focus event

* Shorten to just FocusEvent instead of ReactFocusEvent

* Add another CHANGELOG entry for the typing fix

---------

Co-authored-by: Glen Davies <glen.davies@automattic.com>

youknowriad committed:
Site Editor: Add a fallback template showing the title and content for the post only mode (#56509)

ntsekouras committed:
Dataviews: Update item actions in grid view (#56501)

* Dataviews: Update item actions in grid view

* Use new dropdown

* remove separators and icons

* add `isCompact` prop instead of viewType

getdave committed:
Disable renaming support for Nav Link block (#56425)

juanmaguitar committed:
Merge branch 'fundamentals-block-development/file-structure-of-a-block' of github.com:WordPress/gutenberg into fundamentals-block-development/file-structure-of-a-block

juanmaguitar commented:
Closed in favour of https://github.com/WordPress/gutenberg/pull/56551
